### PR TITLE
Add PostHog token cookie middleware

### PR DESCRIPTION
**Pull request by yakkomajuri**
Adds 2 new cookies that will be used to autofill the user's token in our docs if they're authenticated into Cloud.

- `ph_current_project_token`
- `ph_current_project_name`

Both are needed because given that we now have multiple projects, it'll be important to show the user a tooltip saying what project's token this is.

### CODE REVIEW

File: multi_tenancy/tests/test_middleware.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        self.assertEqual(ph_project_name_cookie["path"], "/")
+        self.assertEqual(ph_project_name_cookie["samesite"], "None")
+        self.assertEqual(ph_project_name_cookie["httponly"], True)
+        self.assertEqual(ph_project_name_cookie["domain"], "")
+        self.assertEqual(ph_project_name_cookie["comment"], "")
+        self.assertEqual(ph_project_name_cookie["secure"], True)
+        self.assertEqual(ph_project_name_cookie["max-age"], 31536000)
+
+    def test_logged_in_client_secure(self):
+        self.client.force_login(self.user)
```
Comment (paolodamico) [inline multi_tenancy/tests/test_middleware.py:43]: This test case seem duplicated

File: multi_tenancy/middleware.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+from django.contrib.sessions.middleware import SessionMiddleware
+
+default_cookie_options = {
+    'max_age': 365 * 24 * 60 * 60,  # one year
+    'expires': None,  
+    'path': "/",  
+    'domain': None,  
+    'secure': True,  
+    'httponly': True,  
+    'samesite': "Lax",  # can't be set to "None" here
```
Comment (paolodamico) [inline multi_tenancy/middleware.py:10]: I think we could do `Strict` here

File: multi_tenancy/middleware.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+from django.contrib.sessions.middleware import SessionMiddleware
+
+default_cookie_options = {
+    'max_age': 365 * 24 * 60 * 60,  # one year
+    'expires': None,  
+    'path': "/",  
+    'domain': None,  
+    'secure': True,  
+    'httponly': True,
```
Comment (paolodamico) [inline multi_tenancy/middleware.py:9]: While this is certainly more secure, wouldn't you need to access the cookie on the client-side to show it on the docs page?

File: multi_tenancy/middleware.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
+from django.contrib.sessions.middleware import SessionMiddleware
+
+default_cookie_options = {
+    'max_age': 365 * 24 * 60 * 60,  # one year
+    'expires': None,  
+    'path': "/",  
+    'domain': None,
```
Comment (paolodamico) [inline multi_tenancy/middleware.py:7]: Maybe worth explicitly setting this for `posthog.com`?

File: multi_tenancy/middleware.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    'samesite': "Lax",  # can't be set to "None" here 
+}
+
+
+class PosthogTokenCookieMiddleware(SessionMiddleware):
+    def process_response(self, request, response):
+        response = super(PosthogTokenCookieMiddleware, self).process_response(request, response)
+
+        # skip adding the cookie on API requests
+        if request.path.startswith("/api/") or request.path.startswith("/e/") or request.path.startswith("/decide/"):
```
Comment (paolodamico) [inline multi_tenancy/middleware.py:19]: Actually, it's important to remove the trailing slash, as the endpoints may or may not have it. In addition, I believe you're at least missing `/batch`, `/track`, `/capture` & `/s`

**Review by paolodamico — CHANGES_REQUESTED** [review]

**Comment by yakkomajuri** [comment]
Great points - a lot of the basics here came from the toolbar cookies and indeed the toolbar needs much more liberal settings than this. Will update!

**Comment by yakkomajuri** [comment]
@paolodamico ready for another go

**Review by paolodamico — APPROVED** [review]

**Pull request merged**